### PR TITLE
[HWKMETRICS-763] fix temp tables not getting dropped

### PR DESCRIPTION
**Pull request by jsanda**

### CODE REVIEW

File: core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/TempTablesCleaner.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    private volatile boolean finished;
+
+    private static final String DROP_TABLE_CQL = "DROP TABLE IF EXISTS %s";
+
+    public TempTablesCleaner(RxSession session, DataAccessImpl dataAccess, String keyspace, int ttl) {
+        this.session = session;
+        this.dataAccess = dataAccess;
+        this.ttl = Days.days(ttl).toStandardDuration().getMillis();
+
+        findTables = session.getSession().prepare(
```
Comment (burmanm) [inline core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/TempTablesCleaner.java]: Other option could be to use the PR #927 findExpiredTables.

Comment (jsanda) [reply core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/TempTablesCleaner.java]: Doesn't findExpiredTables just return timestamps?

Comment (burmanm) [reply core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/TempTablesCleaner.java]: Sure, but for those we have ``dataAccess.dropTempTable(startTimeSlice)``

Comment (burmanm) [reply core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/TempTablesCleaner.java]: mm, actually expired tables is actually just pushing tables that should have been compressed, not dropped completely..

**Comment by jsanda** [comment]
I guess it is a bit confusing that “expired” is used both here and in
DataAccessImpl and refers to different things.

File: core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/TempTablesCleaner.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+    private Observable<String> dropTable(String table) {
+        return session.execute(String.format(DROP_TABLE_CQL, table))
+                .map(resultSet -> table)
+                .onErrorResumeNext(t -> {
+                    // If there is an error, we do not retry because it is possible that the table has already been
+                    // dropped. We will instead wait until findTables runs again and retry dropping the table then
+                    // if dropping it did indeed fail for some reason.
+                    logger.infof(t, "Failed to drop %s", table);
+                    return Observable.just("");
```
Comment (burmanm) [inline core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/TempTablesCleaner.java]: Would ``Observable.empty()`` instead and removing the filter part work as well?

**Comment by jsanda** [comment]
@burmanm I made the changes.

**Pull request merged**